include/oneapi/dpl/internal/common_config.h

```diff
@@ -16,6 +16,14 @@
 #ifndef _ONEDPL_COMMON_CONFIG_H
 #define _ONEDPL_COMMON_CONFIG_H
 
+#if !defined(__cplusplus)
+#    error "oneDPL requires correct state of __cplusplus : it's undefined"
```

Comment:
I can't reproduce this exotic case but seems to mat this check is also required for any cases